syntax = "proto3";

package metadata;

service MetadataService {
  rpc AddSong (AddSongRequest) returns (SongResponse);
  rpc GetSongMetadata (SongRequest) returns (SongResponse);
  rpc GetSongPath (SongRequest) returns (SongPathResponse);
}

message AddSongRequest {
  bytes file_data = 1; // Contenido del archivo de audio
}

message SongRequest {
  string song_id = 1;
}

message SongResponse {
  string song_id = 1;
  string title = 2;
  string artist = 3;
  string album = 4;
  string year = 5;
  string genre = 6;
  double duration = 7;
  bytes album_cover = 8;
}

message SongPathResponse {
  string path = 1;
}
